cmd/webhook/main.go

```diff
@@ -31,78 +31,104 @@ import (
 	ctrlruntimeclient "sigs.k8s.io/controller-runtime/pkg/client"
 )
 
-var (
-	masterURL              string
-	kubeconfig             string
-	admissionListenAddress string
-	admissionTLSCertPath   string
-	admissionTLSKeyPath    string
-	caBundleFile           string
-	useOSM                 bool
-	namespace              string
-)
+type options struct {
+	masterURL               string
+	kubeconfig              string
+	admissionListenAddress  string
+	admissionTLSCertPath    string
+	admissionTLSKeyPath     string
+	caBundleFile            string
+	useOSM                  bool
+	namespace               string
+	workerClusterKubeconfig string
+}
 
 func main() {
 	nodeFlags := node.NewFlags(flag.CommandLine)
+	opt := &options{}
 
 	klog.InitFlags(nil)
 	if flag.Lookup("kubeconfig") == nil {
-		flag.StringVar(&kubeconfig, "kubeconfig", "", "Path to a kubeconfig. Only required if out-of-cluster.")
+		flag.StringVar(&opt.kubeconfig, "kubeconfig", "", "Path to a kubeconfig. Only required if out-of-cluster.")
 	}
 	if flag.Lookup("master") == nil {
-		flag.StringVar(&masterURL, "master", "", "The address of the Kubernetes API server. Overrides any value in kubeconfig. Only required if out-of-cluster.")
+		flag.StringVar(&opt.masterURL, "master", "", "The address of the Kubernetes API server. Overrides any value in kubeconfig. Only required if out-of-cluster.")
 	}
-	flag.StringVar(&admissionListenAddress, "listen-address", ":9876", "The address on which the MutatingWebhook will listen on")
-	flag.StringVar(&admissionTLSCertPath, "tls-cert-path", "/tmp/cert/cert.pem", "The path of the TLS cert for the MutatingWebhook")
-	flag.StringVar(&admissionTLSKeyPath, "tls-key-path", "/tmp/cert/key.pem", "The path of the TLS key for the MutatingWebhook")
-	flag.StringVar(&caBundleFile, "ca-bundle", "", "path to a file containing all PEM-encoded CA certificates (will be used instead of the host's certificates if set)")
-	flag.StringVar(&namespace, "namespace", "kubermatic", "The namespace where the webhooks will run")
+	flag.StringVar(&opt.admissionListenAddress, "listen-address", ":9876", "The address on which the MutatingWebhook will listen on")
+	flag.StringVar(&opt.admissionTLSCertPath, "tls-cert-path", "/tmp/cert/cert.pem", "The path of the TLS cert for the MutatingWebhook")
+	flag.StringVar(&opt.admissionTLSKeyPath, "tls-key-path", "/tmp/cert/key.pem", "The path of the TLS key for the MutatingWebhook")
+	flag.StringVar(&opt.caBundleFile, "ca-bundle", "", "path to a file containing all PEM-encoded CA certificates (will be used instead of the host's certificates if set)")
+	flag.StringVar(&opt.namespace, "namespace", "kubermatic", "The namespace where the webhooks will run")
+	flag.StringVar(&opt.workerClusterKubeconfig, "worker-cluster-kubeconfig", "", "Path to kubeconfig of worker/user cluster where machines and machinedeployments exist")
```

Comment:
Can we please add a note to the description saying that if not passed, will use the `--kubeconfig` kubeconfig? Basically, what is said in the PR description. I think otherwise  the output of `--help` is confusing (which kubeconfig do I need to set if I want to pass only one?)

Reply:
Good point, updated